**Dictionary Programs**

**1)** To perform dictionary methods or actions in the given code snipet.

In [ ]:
inventory = {
    'gold' : 500,
    'pouch' : ['flint', 'twine', 'gemstone'],
    'backpack' : ['xylophone','dagger', 'bedroll','bread loaf']
}
inventory['pocket'] = ['seashell', 'strange berry','lint']
inventory['backpack'].sort()
inventory['pouch'].remove('twine')
print(f"Updated list\n{inventory}")

Updated list
{'gold': 500, 'pouch': ['flint', 'gemstone'], 'backpack': ['bedroll', 'bread loaf', 'dagger', 'xylophone'], 'pocket': ['seashell', 'strange berry', 'lint']}


**Output**\
Updated list\
{'gold': 500, 'pouch': ['flint', 'twine', 'gemstone'], 
\'backpack': ['bedroll', 'bread loaf', 'xylophone'], 
\'pocket': ['seashell', 'strange berry', 'lint']}

**2)** To perform dictionary methods and actions are given in this snipet. 

In [ ]:
prices = {
    'banana':4,
    'apple':2,
    'orange':1.5,
    'pears':3
}
stock = {
    'banana':13,
    'apple':0,
    'orange':7,
    'pears':12
}
for key in prices: 
  print(key)
  print(f"price: {prices[key]}")
  print(f"stock:{stock[key]}")
  print("\n")

totalVal = 0 
comodity = 0 

for key in prices:
  comodity = (prices[key] * stock[key])
  totalVal += comodity
  print(comodity, end='\n\n')
  print(f'Total = {totalVal}')

banana
price: 4
stock:13


apple
price: 2
stock:0


orange
price: 1.5
stock:7


pears
price: 3
stock:12


52

Total = 52
0

Total = 52
10.5

Total = 62.5
36

Total = 98.5


**Output**\
  banana\
price: 4\
stock:13\
\
\
apple\
price: 2\
stock:0\
\
\
orange\
price: 1.5\
stock:7\
\
\
pears\
price: 3\
stock:12\
\
\
52\
\
Total = 52\
0\
\
Total = 52\
10.5\
\
Total = 62.5\
36\
\
Total = 98.5


**3)** To write a python program to calculate the price of comodities. 

In [ ]:
def computeBill(food):
  total = 0 
  for i in food: 
    total += prices[i]
  return total
stock = {
    "banana": 6, 
    "apple": 0, 
    "orange": 32,
    "pear": 15
}

prices = {
    "banana": 4, 
    "apple": 2, 
    "orange": 1.5,
    "pear": 3
}

groceries = ["banana", "orange", "apple"]

computeBill(groceries)

7.5

**Output**\
7.5

**4)** To write a python program to make a teachers grade book. 



In [ ]:
lloyd = {
  "name": "Lloyd",
  "homework": [90.0,97.0,75.0,92.0],
  "quizzes": [88.0,40.0,94.0],
  "tests": [75.0,90.0]
}

alice = {
  "name": "Alice",
  "homework": [100.0, 92.0, 98.0, 100.0],
  "quizzes": [82.0, 83.0, 91.0],
  "tests": [89.0, 97.0]
}

tyler = {
  "name": "Tyler",
  "homework": [0.0, 87.0, 75.0, 22.0],
  "quizzes": [0.0, 75.0, 78.0],
  "tests": [100.0, 100.0]
}

students = [lloyd,alice,tyler]

for student in students:
  print("Name: "+ student["name"])
  print("Scores in Homework: "+ str(student["homework"]))
  print("Scores in quizzes: "+ str(student["quizzes"]))
  print("Scores in tests: "+ str(student["tests"]))
  print("\n")

def average(numbers):
  total = sum(numbers)
  total = float(total)
  return total / len(numbers)

def get_average(student):
  homework = average(student["homework"])
  quizzes = average(student["quizzes"])
  tests = average(student["tests"])
  final = homework*0.1 + quizzes*0.3 + tests*0.6
  return final

def get_letter_grade(score):
  if score >= 90:
      return 'A'
  elif score >=80:
      return 'B'
  elif score >=70:
      return 'C'
  elif score >=60:
      return 'D'
  else:
      return 'F'
      
#get_letter_grade(get_average(lloyd))
def get_class_average(students):
  results = []
  for student in students:
    a = get_average(student)
    results.append(a)
    return average(results)
print("Class Average:", get_class_average(students))
print("Average class grade:", get_letter_grade(get_class_average(students)))

Name: Lloyd
Scores in Homework: [90.0, 97.0, 75.0, 92.0]
Scores in quizzes: [88.0, 40.0, 94.0]
Scores in tests: [75.0, 90.0]


Name: Alice
Scores in Homework: [100.0, 92.0, 98.0, 100.0]
Scores in quizzes: [82.0, 83.0, 91.0]
Scores in tests: [89.0, 97.0]


Name: Tyler
Scores in Homework: [0.0, 87.0, 75.0, 22.0]
Scores in quizzes: [0.0, 75.0, 78.0]
Scores in tests: [100.0, 100.0]


Class Average: 80.55
Average class grade: B


**Output**\

Name: Lloyd\
Scores in Homework: [90.0, 97.0, 75.0, 92.0]\
Scores in quizzes: [88.0, 40.0, 94.0]\
Scores in tests: [75.0, 90.0]\
\
\
Name: Alice\
Scores in Homework: [100.0, 92.0, 98.0, 100.0]\
Scores in quizzes: [82.0, 83.0, 91.0]\
Scores in tests: [89.0, 97.0]\
\
\
Name: Tyler\
Scores in Homework: [0.0, 87.0, 75.0, 22.0]\
Scores in quizzes: [0.0, 75.0, 78.0]\
Scores in tests: [100.0, 100.0]\
\
\
Class Average: 80.55\
Average class grade: B